bolt-taproot-gossip.md

```diff
@@ -0,0 +1,748 @@
+fields. Timestamp fields are also updated to be block heights instead of Unix
+timestamps.
+
+## Terminology
+
+- Collectively, the set of new gossip messages will be referred to as 
+  `taproot gossip`
+- `node_1` and `node_2` refer to the two parties involved in opening a channel.
+- `node_ID_1` and `node_ID_2` respectively refer to the public keys that
+  `node_1` and `node_2` use to identify their nodes in the network.
+- `bitcoin_key_1` and `bitcoin_key_2` respectively refer to the public keys
+  that `node_1` and `node_2` will use for the specific channel being opened.
+  The funding transactions output will be derived from these two keys.
+
+## Type Definitions
+
+The following convenient types are defined:
+
+* `schnorr_signature`: a 64-byte Schnorr signature as defined
+  in [BIP340][bip-340].
+* `partial_signature`: a 32-byte partial Musig2 signature as defined
+  in [BIP-MuSig2][bip-musig2].
+* `public_nonce`: a 66-byte public nonce as defined in [BIP-MuSig2][bip-musig2].
+
+## TLV Based Messages
+
+The initial set of Lightning Network messages consisted of fields that had to 
+be present. Later on, TLV encoding was introduced which provided a backward 
+compatible way to extend messages. To ensure that the new messages defined in 
+this document remain as future-proof as possible, the messages will be pure 
+TLV streams. 
+
+## Taproot Channel Proof and Verification
+
+Taproot channel funding transaction outputs will be SegWit V1 (P2TR) outputs of
+the following form:
+
+```
+OP_1 <taproot_output_key> 
+```
+
+where 
+```
+taproot_internal_key = MuSig2.KeyAgg(MuSig2.KeySort(bitcoin_key_1, bitcoin_key_2))
+
+taproot_output_key = taproot_internal_key + tagged_hash("TapTweak", taproot_internal_key) * G
+```
+
+`taproot_interal_key` is the aggregate key of `bitcoin_key_1` and
+`bitcoin_key_2` after first sorting the keys using the
+[MuSig2 KeySort][musig-keysort] algorithm and then running the
+[MuSig2 KeyAgg][musig-keyagg] algorithm.
+
+Then, to commit to spending the taproot output via the keyspend path, a
+[BIP86][bip86-tweak] tweak is added to the internal key to calculate the
+`taproot_output_key`.
+
+As with legacy channels, nodes will want to perform some checks before adding 
+an announced taproot channel to their routing graph:
+
+1. The funding output of the channel being advertised exists on-chain, is an
+   unspent UTXO and has a sufficient number of confirmations. To allow this 
+   check, the SCID of the channel will continue to be included in the channel
+   announcement.
+2. The funding transaction is sufficiently bound to the Lightning context. 
+   Nodes will do this by checking that they are able to derive the output key 
+   found on-chain by using the advertised `bitcoin_key_1` and `bitcoin_key_2` 
+   along with a [BIP86 tweak][bip86-tweak]. This provides a slightly weaker
+   binding to the LN context than legacy channels do but at least somewhat 
```

Comment:
Maybe useful to the reader to explain how it is weaker?

Reply:
> You mean in terms of what is visible to the blockchain?

 the proof to other nodes that this output will be used for a channel (and is thus not a fake channel)

Reply:
Isn't it still possible for an outsider to verify that the taproot address pays to the combination of the nodes pubkeys?